query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/condition/ConditionalStatement.java

```diff
@@ -0,0 +1,152 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.querybuilder.condition;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.querybuilder.QueryBuilderDsl;
+import com.datastax.oss.driver.api.querybuilder.term.Term;
+import com.datastax.oss.driver.internal.querybuilder.condition.DefaultConditionBuilder;
+import com.datastax.oss.driver.internal.querybuilder.relation.ColumnComponentLeftHandSide;
+import com.datastax.oss.driver.internal.querybuilder.relation.ColumnLeftHandSide;
+import com.datastax.oss.driver.internal.querybuilder.relation.FieldLeftHandSide;
+import java.util.Arrays;
+
+/**
+ * A statement that can be applied conditionally, such as UPDATE or DELETE.
+ *
+ * <p>Not that this does not covert INSERT... IF NOT EXISTS, which is handled separately.
+ */
+public interface ConditionalStatement<SelfT extends ConditionalStatement<SelfT>> {
+
+  /**
+   * Adds an IF EXISTS condition.
+   *
+   * <p>If any column conditions were added before, they will be cleared.
+   */
+  SelfT ifExists();
+
+  /**
+   * Adds an IF condition. All conditions are logically joined with AND. If {@link #ifExists()} was
+   * invoked on this statement before, it will get cancelled.
+   *
+   * <p>To create the argument, use one of the {@code ifXxx} factory methods in {@link
+   * QueryBuilderDsl}, for example {@link QueryBuilderDsl#ifColumn(CqlIdentifier) isColumn}.
+   *
+   * <p>If you add multiple conditions as once, consider {@link #if_(Iterable)} as a more efficient
+   * alternative.
+   */
+  SelfT if_(Condition condition);
+
+  /**
+   * Adds multiple IF conditions at once. All conditions are logically joined with AND. If {@link
+   * #ifExists()} was invoked on this statement before, it will get cancelled.
+   *
+   * <p>This is slightly more efficient than adding the relations one by one (since the underlying
+   * implementation of this object is immutable).
+   *
+   * <p>To create the arguments, use one of the {@code ifXxx} factory methods in {@link
+   * QueryBuilderDsl}, for example {@link QueryBuilderDsl#ifColumn(CqlIdentifier) isColumn}.
+   */
+  SelfT if_(Iterable<Condition> conditions);
+
+  /** Var-arg equivalent of {@link #if_(Iterable)}. */
+  default SelfT if_(Condition... conditions) {
+    return if_(Arrays.asList(conditions));
+  }
+
+  /**
+   * Adds an IF condition on a simple column, as in {@code DELETE... IF k=1}.
+   *
+   * <p>This is the equivalent of creating a condition with {@link
+   * QueryBuilderDsl#ifColumn(CqlIdentifier)} and passing it to {@link #if_(Condition)}.
+   */
+  default ConditionBuilder<SelfT> ifColumn(CqlIdentifier columnId) {
```

Comment:
Definitely too close to `isColumn`, so that confirms that `isColumn` should be renamed to `whereColumn`.